src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/CommentUtils.java

```diff
@@ -626,7 +626,7 @@ public DocCommentInfo getHtmlCommentInfo(Element e) {
 
     public DocCommentTree parse(URI uri, String text) {
         return trees.getDocCommentTree(new SimpleJavaFileObject(
-                uri, JavaFileObject.Kind.SOURCE) {
+                uri, JavaFileObject.Kind.HTML) {
```

Comment:
I would describe it as a "latent bug, waiting to happen".
Previously, all file objects were regarded as containing HTML content, and so there was no need to use the parameter, although maybe it would have been good to check it.

Now, file objects can be HTML or Markdown, and so we do need to use the parameter.

In the case here, the method is used on strings specified in command-line options, which are specified to be in HTML format.   Yes, we might want to change that, but that would be a different RFE separate from the work in this PR.